### PR TITLE
Add git merge to sync both master and poBranch

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

This commit will sync both `master` and `poBranch` so that the translated files are in sync with the source .rst files (english). 

Referencing issue #475

### CODE REVIEW

**Comment by SooluThomas** [comment]
test didn't work because the branch was made in my repo

File: tools/deploy_documentation.sh [hunk 1]
```diff
@@ -62,6 +48,31 @@ git branch
 git config user.name "Qiskit Autodeploy"
 git config user.email "qiskit@qiskit.org"
 
+git checkout master
+git fetch origin/master
```
Comment (mtreinish) [inline tools/deploy_documentation.sh]: This isn't a valid command, `origin/master` is not a remote so you can't fetch it.

File: tools/deploy_documentation.sh [hunk 2]
```diff
@@ -26,22 +26,8 @@ SOURCE_REPOSITORY="git@github.com:Qiskit/qiskit.git"
...
-# Extract document's translatable messages into pot files
-# https://sphinx-intl.readthedocs.io/en/master/quickstart.html
-echo "Extract document's translatable messages into pot files: "
-sphinx-build -b gettext -D language=$TRANSLATION_LANG . _build/gettext
-
-# Setup / Update po files
-echo "Setup / Update po files"
-sphinx-intl update -p _build/gettext -l en
+echo "git branch at the beginning"
+git branch
```
Comment (mtreinish) [inline tools/deploy_documentation.sh]: Why are you listing all the branches at the beginning?

**Review by mtreinish — CHANGES_REQUESTED** [review]
I'm confused as to the purpose of this change, why do you need to merge master into poBranch? Isn't poBranch just where the generated .po files live. Shouldn't we just generate the English strings from master copy those into poBranch and push an update from that. What does merging master accomplish on top of that?

Additionally, trying to do a merge will have conflicts which must be resolved manually. Git is good at merging things automatically (using things like 3 way merge, etc), but there are a lot of cases where it can't resolve a conflict and manual intervention is required. This is especially true when the branches have diverged like master and poBranch have. This will fail as soon as you run it because there are merge conflicts (go ahead and try `git checkout origin/poBranch && git merge origin/master` in your local repo to see).

That all being said I found an issue inline too.

File: tools/deploy_documentation.sh (hunk 2 above)
Comment (SooluThomas) [reply tools/deploy_documentation.sh]: to check what branch it is in at the beginning of the build. I found the answer by running a test build in my fork. I think I can remove `git branch`.

**Comment by SooluThomas** [comment]
> I'm confused as to the purpose of this change, why do you need to merge master into poBranch? Isn't poBranch just where the generated .po files live. Shouldn't we just generate the English strings from master copy those into poBranch and push an update from that. What does merging master accomplish on top of that?

Both `master` and `poBranch` have to be in sync because, `ja` translations are built from both `.po` files and `.rst` files. Which requires updated `.rst` file in both `master` and `poBranch`.
(Either manual or automated).

For example, if there are any changes made to the [theme/layouts](https://github.com/Qiskit/qiskit/tree/master/docs/theme) in `master` branch, they won't be reflected in the finally generated translated html because the "Translated HTML" is generated from both the `.po` files and the `.rst` files in the repo/branch (wherever `sphinx-build -b html -D language=$TRANSLATION_LANG . _build/html/locale/$TRANSLATION_LANG` is run)

Here's a diagram from [Sphinx Internationalization Documentation](https://www.sphinx-doc.org/en/1.8/intl.html)
translation

Do let me know your views on this. Thank you.

File: tools/deploy_documentation.sh (hunk 1 above)
Comment (SooluThomas) [reply tools/deploy_documentation.sh]: Thanks for pointing out the error. I'll rectify it.

**Comment by mtreinish** [comment]
That diagram doesn't explain why you need to merge master into poBranch. In fact it doesn't mention branches anywhere, all it says is that the po and rst files are an input to sphinx build with a translated build. There is nothing that says this has to be in the same branch or even the same tree. Just that you need both.

The way things are setup now you should treat master as the source for the raw source (rst, layout etc) and poBranch as the source of the translated strings. Then when you build documentation (wherever you end up running that) you just pull from both sources. As I said earlier trying to merge master into branch automatically will not work and just prone to failure when there is divergence in the tree (which there already is). Things are actually already setup well to do this, on merges to master we generate new .pot files, update the .po files, and pushes that to poBranch already. That's all you need.

Then docs builds are as simple as something like::

```bash
git clone github.com/Qiskit/qiskit docs_source
git clone github.com/Qiskit/qiskit -b translations translations
cp -r translations/docs/locale docs_source/docs/.
sphinx-build ... docs_source/docs docs_source/_build
```
(although you don't actually need one of the clones because travis clones the repo for us)

**Comment by SooluThomas** [comment]
That's a better idea. Thank you. I'll check if it can work that way.

**Comment by SooluThomas** [comment]
Closing this issue because it's no longer valid

**Pull request closed without merging**